tests/testthat/test-proxy-restore.R

```diff
@@ -129,3 +129,43 @@ test_that("names<- is not called with partial data (#1108)", {
   vec_c(x, x)
   expect_equal(values, list(c("a", "b", "a", "b")))
 })
+
+test_that("proxy and restore pass `recurse` through `...`", {
```

Comment:
So I guess we don't want to automatically apply this to rcrd vectors? i.e. in

```r
> vctrs:::vec_proxy.vctrs_rcrd
function(x, ...) {
  new_data_frame(x)
}
```

And the corresponding restore method

I'd still have to do it in ivs anyways because I have custom methods, but I'm not sure if proxy/restore should automatically be recursive here or not